pkg/controller/kubefedcluster/controller_test.go

```diff
@@ -65,17 +65,17 @@ func TestThresholdCheckedClusterStatus(t *testing.T) {
 		"ClusterNotReadyButWithinFailureThreshold": {
 			clusterStatus: clusterStatus(corev1.ConditionFalse, t3, t3),
 			storedClusterData: &ClusterData{
-				clusterStatus: clusterStatus(corev1.ConditionTrue, t2, t1),
-				resultRun:     2},
-			expectedClusterStatus: clusterStatus(corev1.ConditionTrue, t3, t1),
-			expectedResultRun:     3,
+				clusterStatus: clusterStatus(corev1.ConditionFalse, t2, t1),
```

Comment:
This is a valid test that you are changing. The test checks that the cluster is currently ready, but cluster status check is saying that there is a problem with the cluster. It additionally states that the check has happened two times before and this is the third time we are checking. The result is as expected with existing code and existing test. The cluster status should still remain `ready=true`.

With your code change this test is failing.

Reply:
@irfanurrehman  thanks a lot for your detailed explanation and I think your suggestion is very good.